cookbooks/fb_apache/recipes/default.rb

```diff
@@ -106,6 +106,27 @@
   end
 end
 
+if node.debian? || node.ubuntu?
+  # CentOS makes this symlink to the right module dir, and we make assumptions
+  # it exists, so be sure to do the same on debian
+  link '/etc/apache2/modules' do
+    to '/usr/lib/apache2/modules'
+  end
+
+  # For reasons I don't understand on Ubuntu, Apache looks for mime.types in
+  # /etc/apache2/mime.types even though it's not configured to. So make a
+  # symlink
+  link '/etc/apache2/mime.types' do
+    to '/etc/mime.types'
+  end
+end
+
+# The package comes pre-installed with module configs, but we dropp off our own
+# in fb_modules.conf. Also, we don't want non-Chef controlled module configs.
+fb_apache_cleanup_modules 'doit' do
+  mod_dir moddir
+end
+
```

Comment:
@jaininshah9 tested this on some fbit hosts and found we're dropping a lot of modules part of the Apache default config, which breaks stuf (e.g.: `mod_rewrite`). We could (and likely should) add those on our end, but I think we should at double check and update the default here to match the distro default.

Reply:
I picked a default that lets the distro-default service start. Keeping up with whatever modules the distro disables/enables by default over time is probably a bit of a tall order, but I don't mind adding some...

Also keep in mind though that it's more intuitive for people to add modules than to remove them. And that additional modules can mean additional security vulns. So it seems to me less is more here?

Reply:
@jaininshah9 can you share the list you have on your hosts, so we can make an informed decision? I tend to agree that we probably shouldn't add everything, but if we're missing something glaring we should fix it.

Reply:
@davide125 - yeah that seems like a good idea. For rewrite in particular, I think it should not be in the default - it's something you have to turn-on per vhost anyway, so it seems like adding that module at the same time makes sense.

Reply:
Here is the list of enabled loaded modules (sorry for the long post, didn't know how else to post as paste wouldn't work here)

```
Loaded Modules:
 core_module (static)
 so_module (static)
 http_module (static)
 access_compat_module (shared)
 actions_module (shared)
 alias_module (shared)
 allowmethods_module (shared)
 auth_basic_module (shared)
 auth_digest_module (shared)
 authn_anon_module (shared)
 authn_core_module (shared)
 authn_dbd_module (shared)
 authn_dbm_module (shared)
 authn_file_module (shared)
 authn_socache_module (shared)
 authz_core_module (shared)
 authz_dbd_module (shared)
 authz_dbm_module (shared)
 authz_groupfile_module (shared)
 authz_host_module (shared)
 authz_owner_module (shared)
 authz_user_module (shared)
 autoindex_module (shared)
 cache_module (shared)
 cache_disk_module (shared)
 data_module (shared)
 dbd_module (shared)
 deflate_module (shared)
 dir_module (shared)
 dumpio_module (shared)
 echo_module (shared)
 env_module (shared)
 expires_module (shared)
 ext_filter_module (shared)
 filter_module (shared)
 headers_module (shared)
 include_module (shared)
 info_module (shared)
 log_config_module (shared)
 logio_module (shared)
 mime_magic_module (shared)
 mime_module (shared)
 negotiation_module (shared)
 remoteip_module (shared)
 reqtimeout_module (shared)
 rewrite_module (shared)
 setenvif_module (shared)
 slotmem_plain_module (shared)
 slotmem_shm_module (shared)
 socache_dbm_module (shared)
 socache_memcache_module (shared)
 socache_shmcb_module (shared)
 status_module (shared)
 substitute_module (shared)
 suexec_module (shared)
 unique_id_module (shared)
 unixd_module (shared)
 userdir_module (shared)
 version_module (shared)
 vhost_alias_module (shared)
 dav_module (shared)
 dav_fs_module (shared)
 dav_lock_module (shared)
 lua_module (shared)
 mpm_prefork_module (shared)
 proxy_module (shared)
 lbmethod_bybusyness_module (shared)
 lbmethod_byrequests_module (shared)
 lbmethod_bytraffic_module (shared)
 lbmethod_heartbeat_module (shared)
 proxy_ajp_module (shared)
 proxy_balancer_module (shared)
 proxy_connect_module (shared)
 proxy_express_module (shared)
 proxy_fcgi_module (shared)
 proxy_fdpass_module (shared)
 proxy_ftp_module (shared)
 proxy_http_module (shared)
 proxy_scgi_module (shared)
 proxy_wstunnel_module (shared)
 xml2enc_module (shared)
 proxy_html_module (shared)
 ssl_module (shared)
 systemd_module (shared)
 cgi_module (shared)
```

Reply:
Thanks @jaininshah9. FYI you can use https://gist.github.com/ to make pastes for github.